File: src/HomeBudget.API/Models/Transaction/TransactionStatistics.cs
using HomeBudget.API.Entities;

namespace HomeBudget.API.Models.Transaction;

public class TransactionStatistics
{
    public TransactionStatistics(User user)
    {
        MonthlyTotal = new decimal[12];
        User = user;
    }

    public User User { get; set; }
    public decimal[] MonthlyTotal { get; set; }
}

File: src/homebudget-frontend/src/entities/TransactionStatistics.ts
import User from "./User";

export default class TransactionStatistics {
    private _user: User;
    private _monthlyTotal: number[];
    
    constructor(user: User, monthlyTotal: number[]) {
        this._user = user;
        this._monthlyTotal = monthlyTotal;
    }
    
    get user(): User {
        return this._user;
    }
    
    set user(value: User) {
        this._user = value;
    }
    
    get monthlyTotal(): number[] {
        return this._monthlyTotal;
    }
    
    set monthlyTotal(value: number[]) {
        this._monthlyTotal = value;
    }
}